src/test/java/org/opentripplanner/middleware/ApiUserFlowTest.java

```diff
@@ -150,32 +162,37 @@ public void canSimulateApiUserFlow() throws URISyntaxException {
         // Create a monitored trip for the Otp user (API users are prevented from doing this).
         MonitoredTrip monitoredTrip = new MonitoredTrip(TestUtils.sendSamplePlanRequest());
         monitoredTrip.userId = otpUser.id;
-        HttpResponse<String> createTripResponseAsOtpUser = mockAuthenticatedPost("api/secure/monitoredtrip",
+        HttpResponse<String> createTripResponseAsOtpUser = mockAuthenticatedRequest("api/secure/monitoredtrip",
+            JsonUtils.toJson(monitoredTrip),
             otpUserResponse,
-            JsonUtils.toJson(monitoredTrip)
+            HttpUtils.REQUEST_METHOD.POST,
+            true
         );
         assertEquals(HttpStatus.UNAUTHORIZED_401, createTripResponseAsOtpUser.statusCode());
 
         // Create a monitored trip for an Otp user authenticating as an Api user. An Api user can create a monitored
         // trip for an Otp user they created.
-        HttpResponse<String> createTripResponseAsApiUser = mockAuthenticatedPost("api/secure/monitoredtrip",
-            apiUser,
-            JsonUtils.toJson(monitoredTrip)
+        HttpResponse<String> createTripResponseAsApiUser = authenticatedRequest("api/secure/monitoredtrip",
+            JsonUtils.toJson(monitoredTrip),
+            headers,
+            HttpUtils.REQUEST_METHOD.POST
         );
         assertEquals(HttpStatus.OK_200, createTripResponseAsApiUser.statusCode());
         MonitoredTrip monitoredTripResponse = JsonUtils.getPOJOFromJSON(createTripResponseAsApiUser.body(), MonitoredTrip.class);
 
         // Request all monitored trip for an Otp user authenticating as an Api user.
-        HttpResponse<String> getAllMonitoredTripsForOtpUser = mockAuthenticatedRequest(String.format("api/secure/monitoredtrip?userId=%s",
+        HttpResponse<String> getAllMonitoredTripsForOtpUser = authenticatedRequest(String.format("api/secure/monitoredtrip?userId=%s",
             otpUserResponse.id),
-            apiUser,
+            "",
+            headers,
             HttpUtils.REQUEST_METHOD.GET
```

Comment:
A helper method to mockAuthenticatedGet might be useful since there are a lot of these requests that just pass an empty string for the body.

Reply:
Updated.